common/include/window.h

```diff
@@ -50,6 +50,79 @@ static inline void set_embedded_window_pointer(embed_window_pointer_t *wp, windo
 
 static inline void set_embedded_window_pointer(ignore_window_pointer_t *, window *) {}
 
+struct window
+{
+private:
+	grs_canvas w_canv;					// the window's canvas to draw to
+	window_event_result (*w_callback)(window *wind,const d_event &event, void *data);	// the event handler
+	int w_visible;						// whether it's visible
+	int w_modal;						// modal = accept all user input exclusively
+	void *data;							// whatever the user wants (eg menu data for 'newmenu' menus)
+	struct window *prev;				// the previous window in the doubly linked list
+	struct window *next;				// the next window in the doubly linked list
+	
+public:
+	// Declaring as friends to keep function syntax, for historical reasons (for now at least)
+	friend window *window_create(grs_canvas *src, int x, int y, int w, int h, window_subfunction<void> event_callback, void *userdata, const void *createdata);
+	
+	friend int window_close(window *wind);
+	friend int window_exists(window *wind);
+	friend window *window_get_front();
+	friend window *window_get_first();
+	friend void window_select(window &wind);
+	friend window *window_set_visible(window &wind, int visible);
+#if !DXX_USE_OGL
+	friend void window_update_canvases();
+#endif
+	friend window_event_result window_send_event(window &wind,const d_event &event);
+	friend void window_set_modal(window &wind, int modal);
+	friend int window_is_modal(window &wind);
+	
+	friend grs_canvas &window_get_canvas(window &wind)
+	{
+		return wind.w_canv;
+	}
+	
+	friend window *window_set_visible(window *wind, int visible)
+	{
+		return window_set_visible(*wind, visible);
+	}
+
+	friend int window_is_visible(window *wind)
```

Comment:
Passing a valid `dcx::window` is not optional for `window_is_visible`, so I would prefer to retain the signature `int (dcx::window &)` to enforce this.  A transitional wrapper that permits a pointer and chains to the reference form is fine, but this commit seems to have only the pointer form and to adjust callers that previously called the reference form.